docs/language-spec.md

```diff
@@ -1234,29 +1241,34 @@ even if the value is an empty list literal.
-| `uint32`, `uint64`, `fixed32`, `fixed64`                     | _UintLiteral_ †                                             |
-| `float`, `double`                                            | _FloatLiteral_, _UintLiteral_, _IntLiteral_, _identifier_ ‡ |
-| `bool`                                                       | _identifier_ §                                              |
-| `string`, `bytes`                                            | _StringLiteral_                                             |
-| An enum type                                                 | _identifier_, _UintLiteral_ ¶, _IntLiteral_ ¶               |
-| A message type                                               | _MessageLiteral_                                            |
-
-__*__ Integer field types can only use _UintLiteral_ and _IntLiteral_ values if the
-represented number is in the range [-2<sup>31</sup>,2<sup>31</sup>) for 32-bit types
+| Field Type                                                   | Allowed Value Types                                                |
+|--------------------------------------------------------------|--------------------------------------------------------------------|
+| `int32`, `int64`, `sint32`, `sint64`, `sfixed32`, `sfixed64` | _IntLiteral_ \*                                                    |
+| `uint32`, `uint64`, `fixed32`, `fixed64`                     | _IntLiteral_ †                                                     |
+| `float`, `double`                                            | _FloatLiteral_, _IntLiteral_, _SignedIdentifier_ ‡, _identifier_ ‡ |
+| `bool`                                                       | _identifier_ §                                                     |
+| `string`, `bytes`                                            | _StringLiteral_                                                    |
+| An enum type                                                 | _identifier_, _IntLiteral_ ¶                                       |
+| A message type                                               | _MessageLiteral_                                                   |
+
+__*__ Integer field types can only use _IntLiteral_ values if the represented
+number is in the range [-2<sup>31</sup>,2<sup>31</sup>) for 32-bit types or
 or [-2<sup>63</sup>,2<sup>63</sup>) for 64-bit types.
 
-__†__ Unsigned integer field types can only use _UintLiteral_ values if the
+__†__ Unsigned integer field types can only use _IntLiteral_ values if the
 represented number is in the range [0,2<sup>32</sup>) for 32-bit types
 or [0,2<sup>64</sup>) for 64-bit types.
 
 __‡__ Floating point field types can only use _identifier_ values if the identifier
-is `inf` or `nan`.
+is `inf`, `infinity`, or `nan`. But this is a case-insensitive check (so `INF`
+and `NaN` could also be used). Similarly, the _identifier_ component of a
```

Comment:
I can't get a differently cased nan or inf or infinity to work using buf:
![image](https://github.com/bufbuild/protobuf.com/assets/4567082/6b71f5ad-11eb-49d3-b7ff-149b9bfbacf7)

Reply:
That is unfortunately expected (for now). https://github.com/bufbuild/protocompile/issues/225